### PR TITLE
fix(python): raise if DF is input to context (with_columns, select, etc)

### PR DESCRIPTION
**Pull request by deanm0000**
addresses https://github.com/pola-rs/polars/issues/15588

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
nice - fancy adding a little test too?

**Comment by deanm0000** [comment]
A test to verify that it errors?

**Comment by MarcoGorelli** [comment]
Yup, you can use the pytest.raises context manager

**Comment by deanm0000** [comment]
well that was a good callout b/c there was a way that is snuck by that is now fixed.

File: py-polars/polars/_utils/parse_expr_input.py [hunk 1]
```diff
@@ -121,6 +121,12 @@ def parse_as_expression(
     elif isinstance(input, list) and not list_as_lit:
         expr = F.lit(pl.Series(input), dtype=dtype)
         structify = False
+    elif isinstance(input, pl.DataFrame):
+        msg = "A DataFrame is an invalid input"
```
Comment (ritchie46) [inline py-polars/polars/_utils/parse_expr_input.py:125]:
```suggestion
        msg = "a DataFrame is an invalid input"
```

File: py-polars/polars/_utils/parse_expr_input.py [hunk 2]
```diff
@@ -121,6 +121,12 @@ def parse_as_expression(
...
+        raise InvalidOperationError(msg)
+    elif isinstance(input, pl.LazyFrame):
+        msg = "A LazyFrame is an invalid input"
```
Comment (ritchie46) [inline py-polars/polars/_utils/parse_expr_input.py:128]:
```suggestion
        msg = "a LazyFrame is an invalid input"
```

We don't want to capitalize the first line of an error message.

File: py-polars/polars/_utils/parse_expr_input.py [hunk 3]
```diff
@@ -121,6 +121,12 @@ def parse_as_expression(
     elif isinstance(input, list) and not list_as_lit:
         expr = F.lit(pl.Series(input), dtype=dtype)
         structify = False
+    elif isinstance(input, pl.DataFrame):
```
Comment (stinodego) [inline py-polars/polars/_utils/parse_expr_input.py:124]: This is not the right place to add this. It should be added in the `lit` function.

**Review by stinodego — CHANGES_REQUESTED** [review]

**Comment by stinodego** [comment]
Maybe this should be parsed as a struct column instead of raising?

**Comment by deanm0000** [comment]
> Maybe this should be parsed as a struct column instead of raising?

If they want a struct they can just `df.to_struct()`. I have three reasons for thinking it's not a great idea to do this for people behind the scenes:

1. If they're putting one df as the input to another df's with_columns/select then they're doing an anti-pattern and they should go ask for help so the error is the best thing for them
2. People are likely to be more befuddled by getting a struct where they once got individual columns so it'll feel more like changing behavior without notice than fixing the bug that let them accidentally do the thing.
3. It's a very pandas thing to do to try to anticipate and fix the way people do things incorrectly.

I was scratching my head on suggestions to include in the error message. I was thinking of "hey maybe you need a join or an hstack or to just refactor generally" but there are too many possibilities of what they're doing wrong that I just went with the short one that you see.

**Comment by stinodego** [comment]
Please see the linked issue. This should not raise.

**Pull request closed without merging**